numpy/testing/_private/utils.py

```diff
@@ -164,7 +164,7 @@ def memusage(processName="python", instance=0):
                                         win32pdh.PDH_FMT_LONG, None)
 elif sys.platform[:5] == 'linux':
 
-    def memusage(_proc_pid_stat=f'/proc/{os.getpid()}/stat'):
+    def memusage(_proc_pid_stat=f'/proc/{os.getpid()}/stat'):  # noqa: B008
```

Comment:
This smells like potential source of `multiprocessing`-related bugs to me 🧐. So unless this is performance-critical (which isn't the case, judging by the function body), maybe we should listen to ruff here?

Reply:
Indeed, the default value is initialised once when the function is defined using the PID of the initial process. When the initial process is spawned, the PID of the initial process would be reused, wouldn't it? SO it's best to use `None` and call `os.getpid()}` each time the function is called.

Reply:
> Unless we want to always access the initial process? I mean, I don't know what `memusage` is supposed to do.

Maybe this even behaves differently depending on whether you're using *fork* or *spawn*. But I don't know enough about it to say for sure 🤷🏻